src/postgres/getClientConfig.ts

```diff
@@ -12,39 +12,38 @@ import { invalidCredentialsErrorType } from "./tree/PostgresDatabaseTreeItem";
 import { PostgresServerTreeItem } from "./tree/PostgresServerTreeItem";
 
 export async function getClientConfig(treeItem: PostgresServerTreeItem, databaseName: string): Promise<ClientConfig> {
-    let username: string | undefined = treeItem.connectionString.username;
-    let password: string | undefined = treeItem.connectionString.password;
+    let clientConfig: ClientConfig;
+    if (treeItem.azureName) {
+        const username: string | undefined = treeItem.connectionString.username;
+        const password: string | undefined = treeItem.connectionString.password;
 
-    if (!(username && password)) {
-        const credentials = await treeItem.getCredentials();
-        username = nonNullProp(credentials, 'username');
-        password = nonNullProp(credentials, 'password');
-    }
-
-    const ssl: ConnectionOptions = {
-        // Always provide the certificate since it is accepted even when SSL is disabled
-        // Certificate source: https://aka.ms/AA7wnvl
-        ca: BaltimoreCyberTrustRoot
-    };
-
-    if (username && password) {
-        const host = nonNullProp(treeItem.connectionString, 'hostName');
-        const port: number = treeItem.connectionString.port ? parseInt(treeItem.connectionString.port) : postgresDefaultPort;
-        const clientConfig = { user: username, password: password, ssl, host, port, database: databaseName };
-        const client = new Client(clientConfig);
+        const sslAzure: ConnectionOptions = {
+            // Always provide the certificate since it is accepted even when SSL is disabled
+            // Certificate source: https://aka.ms/AA7wnvl
+            ca: BaltimoreCyberTrustRoot
+        };
 
-        // Ensure the client config is valid before returning
-        try {
-            await client.connect();
-            return clientConfig;
-        } finally {
-            await client.end();
+        if ((username && password) || username === 'postgres') {
+            const host = nonNullProp(treeItem.connectionString, 'hostName');
+            const port: number = treeItem.connectionString.port ? parseInt(treeItem.connectionString.port) : postgresDefaultPort;
+            clientConfig = { user: username, password: password, ssl: sslAzure, host, port, database: databaseName };
+        } else {
+            throw {
+                message: localize('mustEnterCredentials', 'Must enter credentials to connect to server.'),
+                code: invalidCredentialsErrorType
+            };
         }
     } else {
-        throw {
-            message: localize('mustEnterCredentials', 'Must enter credentials to connect to server.'),
-            code: invalidCredentialsErrorType
-        };
+        clientConfig = { connectionString: treeItem.connectionString.connectionString };
```

Comment:
I'm getting this now:
<img width="276" alt="Screen Shot 2020-08-28 at 10 51 47 AM" src="https://user-images.githubusercontent.com/11282622/91600293-92d4d800-e91c-11ea-932d-95eafd833e1a.png">

Seems like "pg" has a defaults object, which uses this:
```
  // name of database to connect
  database: process.platform === 'win32' ? process.env.USERNAME : process.env.USER,
```

I'm not sure why they default to the username, and seems like it [already changed](https://github.com/brianc/node-postgres/blob/master/packages/pg/lib/defaults.js#L11) in the latest version of the "pg" package, but I think the fix is for us to change that `defaults.database` to "postgres" (or eventually try out the new version of the "pg" package)

Reply:
Updating "pg" package would cause the same problem... Here is the latest version:
In defaults, `database: undefined,`
In connection-parameters, `if (this.database === undefined) {  this.database = this.user }`
That said the latest version is more easier to handle. What is the best place for `default.database = 'postgres'` to go, I am not sure where to place it in the general or 'utils' folder?

Reply:
> What is the best place for default.database = 'postgres' to go, I am not sure where to place it in the general or 'utils' folder?

Well we only need to do it once per session of VS Code. So maybe just in the extension.ts/activate method. Or in `registerPostgresCommands` if you want it in a postgres-specific place